engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/hydration/batch/NadelBatchHydrationInputBuilder.kt

```diff
@@ -36,40 +37,47 @@ internal object NadelBatchHydrationInputBuilder {
     private fun getNonBatchInputValues(
         instruction: NadelBatchHydrationFieldInstruction,
         hydrationField: ExecutableNormalizedField,
-    ): Map<NadelHydrationActorInputDef, NormalizedInputValue> {
-        return mapFrom(
+    ): Map<NadelHydrationActorInputDef, CountedBox<NormalizedInputValue>> {
+        var counter = 0
+        val map = mapFrom(
             instruction.actorInputValueDefs.mapNotNull { actorFieldArg ->
                 when (val valueSource = actorFieldArg.valueSource) {
                     is NadelHydrationActorInputDef.ValueSource.ArgumentValue -> {
                         when (val argValue = hydrationField.normalizedArguments[valueSource.argumentName]) {
                             null -> null
-                            else -> actorFieldArg to argValue
+                            else -> {
+                                counter++
+                                actorFieldArg to argValue
+                            }
                         }
                     }
                     // These are batch values, ignore them
                     is NadelHydrationActorInputDef.ValueSource.FieldResultValue -> null
                 }
             },
         )
+        return map.mapValues { CountedBox(it.value, counter) }
     }
 
     private fun getBatchInputValues(
         instruction: NadelBatchHydrationFieldInstruction,
         parentNodes: List<JsonNode>,
         aliasHelper: NadelAliasHelper,
-    ): List<Pair<NadelHydrationActorInputDef, NormalizedInputValue>> {
+    ): List<Pair<NadelHydrationActorInputDef, CountedBox<NormalizedInputValue>>> {
```

Comment:
this conflicts now with the changes that I made to support multi tenanted hydrations
I think it should be something like
```kotlin
private fun getBatchInputValues(
        instruction: NadelBatchHydrationFieldInstruction,
        parentNodes: List<JsonNode>,
        aliasHelper: NadelAliasHelper,
        hooks: ServiceExecutionHooks,
    ): List<Pair<NadelHydrationActorInputDef, CountedBox<NormalizedInputValue>>> {
        val batchSize = instruction.batchSize

        val (batchInputDef, batchInputValueSource) = getBatchInputDef(instruction) ?: return emptyList()
        val actorBatchArgDef = instruction.actorFieldDef.getArgument(batchInputDef.name)

        val fieldResultValues = getFieldResultValues(batchInputValueSource, parentNodes, aliasHelper)

        val partitionedArgumentList = when (hooks) {
            is NadelEngineExecutionHooks -> hooks.partitionBatchHydrationArgumentList(fieldResultValues, instruction)
            else -> listOf(fieldResultValues)
        }

        return partitionedArgumentList.flatMap { it.chunked(size = batchSize) }
            .map { chunkedFieldResultValues ->
                val normalizedInputValue = NormalizedInputValue(
                    GraphQLTypeUtil.simplePrint(actorBatchArgDef.type),
                    javaValueToAstValue(chunkedFieldResultValues),
                )
                batchInputDef to CountedBox(normalizedInputValue, chunkedFieldResultValues.size)
            }
    }
```